servers/src/main/java/tachyon/master/file/FileSystemMaster.java

```diff
@@ -0,0 +1,1063 @@
+   */
+  public List<TachyonURI> getInMemoryFiles() {
+    List<TachyonURI> ret = new ArrayList<TachyonURI>();
+    LOG.info("getInMemoryFiles()");
+    Queue<Pair<InodeDirectory, TachyonURI>> nodesQueue =
+        new LinkedList<Pair<InodeDirectory, TachyonURI>>();
+    synchronized (mInodeTree) {
+      // TODO: Verify we want to use absolute path.
+      nodesQueue.add(new Pair<InodeDirectory, TachyonURI>(mInodeTree.getRoot(),
+          new TachyonURI(TachyonURI.SEPARATOR)));
+      while (!nodesQueue.isEmpty()) {
+        Pair<InodeDirectory, TachyonURI> tPair = nodesQueue.poll();
+        InodeDirectory tDir = tPair.getFirst();
+        TachyonURI curUri = tPair.getSecond();
+
+        Set<Inode> children = tDir.getChildren();
+        for (Inode tInode : children) {
+          TachyonURI newUri = curUri.join(tInode.getName());
+          if (tInode.isDirectory()) {
+            nodesQueue.add(new Pair<InodeDirectory, TachyonURI>((InodeDirectory) tInode, newUri));
+          } else if (isFullyInMemory((InodeFile) tInode)) {
+            ret.add(newUri);
+          }
+        }
+      }
+    }
+    return ret;
+  }
+
+  /**
+   * Get the in-memory percentage of an Inode. For a file that has all blocks in memory, it returns
+   * 100; for a file that has no block in memory, it returns 0. Returns 0 for a directory.
+   *
+   * @param inode the inode
+   * @return the in memory percentage
+   */
+  private int getInMemoryPercentage(Inode inode) {
+    if (!inode.isFile()) {
+      return 0;
+    }
+    InodeFile inodeFile = (InodeFile) inode;
+
+    long length = inodeFile.getLength();
+    if (length == 0) {
+      return 100;
+    }
+
+    long inMemoryLength = 0;
+    for (BlockInfo info : mBlockMaster.getBlockInfoList(inodeFile.getBlockIds())) {
+      if (isInMemory(info)) {
+        inMemoryLength += info.getLength();
+      }
+    }
+    return (int) (inMemoryLength * 100 / length);
+  }
+
+  /**
+   * @return true if the given block is in some worker's memory, false otherwise
+   */
+  private boolean isInMemory(BlockInfo blockInfo) {
+    for (BlockLocation location : blockInfo.getLocations()) {
+      if (location.getTier() == StorageLevelAlias.MEM.getValue()) {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  /**
+   * Creates a directory for a given path. Called via RPC, and internal masters.
+   *
+   * @param path the path of the directory
+   * @param recursive if it is true, create necessary but nonexistent parent directories, otherwise,
+   *        the parent directories must already exist
+   * @throws InvalidPathException when the path is invalid, please see documentation on
+   *         {@link InodeTree#createPath} for more details
+   * @throws FileAlreadyExistException when there is already a file at path
+   */
+  public void mkdirs(TachyonURI path, boolean recursive) throws InvalidPathException,
+      FileAlreadyExistException {
+    // TODO: metrics
+    synchronized (mInodeTree) {
```

Comment:
I feel like the style is to have the space. For example, while (), if (), for (). synchronized () seems consistent with that.